src/unlabel.ml

```diff
@@ -0,0 +1,192 @@
+let slurp_file file =
+  let ch = open_in file in
+  let buf = Buffer.create (min 1024 (in_channel_length ch)) in
+  try
+    while true do Buffer.add_channel buf ch 4096 done;
+    assert false
+  with
+    | End_of_file ->
+      close_in ch;
+      Bytes.unsafe_to_string (Buffer.to_bytes buf)
+;;
+
+let () =
+  assert (Array.length Sys.argv = 3);
+  let fmt = Format.err_formatter in
+  let   labelled_mli_filename = Sys.argv.(1) in
+  let unlabelled_mli_filename = Sys.argv.(2) in
+  (*let unlabelled_ml_filename = Sys.argv.(3) in*)
+  let labelled_name = (* e.g. ArrayLabels *)
+    Compenv.module_of_filename fmt
+      labelled_mli_filename
+      labelled_mli_filename
+    |> Str.(replace_first (regexp "^CC") "")
+  in
+  let unlabelled_name = Str.(replace_first (regexp "Labels") "" labelled_name) in
+  let mli_text = slurp_file labelled_mli_filename in
+  let lexbuf = Lexing.from_string mli_text in
+  Location.init lexbuf labelled_mli_filename;
+  let labelled_ast = Parse.interface lexbuf in
+  let labelled_functions = ref [] in
+  let rec iterator =
+    let open Ast_iterator in
+    let open Parsetree in
+    { Ast_iterator.default_iterator with
+        value_description = fun iterator vdesc ->
```

Comment:
We should also remove attributes for `ocaml <= 4.02`, I think. There's a `attributes` field in the iterator where we could filter the attributes specific to this PR (and possibly others like `[@inline]`?).

Reply:
unlabelled.ml does no preprocess *ASTs* like a ppx preprocessor, but *.mli*.
At least at the moment it is only supposed to generate unlabelled .mli from a labelled one. The labelled one is not changed. Would you like a preprocessor reworking all *.mli* and possibly also *.ml* ?

Reply:
I'm afraid the AST won't help with removing attributes from mli, since the AST doesn't contain locations of the attributes.

Reply:
It's not necessarily a good idea to get a general preprocessor right now, I was just thinking forward.
Do you know the difference between `Ast_iterator` and `Ast_mapper`?

Reply:
``Ast_iterator`` is an iterator over the AST while ``Ast_mapper`` is a mapping function mapping for transforming ASTs